xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-api/src/main/java/org/xwiki/livedata/LiveDataPropertyDescriptor.java

```diff
@@ -126,6 +131,41 @@ public void initialize()
                 this.operators = new ArrayList<>();
             }
         }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) {
+                return true;
+            }
+
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            FilterDescriptor that = (FilterDescriptor) o;
+
+            return new EqualsBuilder().appendSuper(super.equals(o))
+                .append(defaultOperator, that.defaultOperator).append(operators, that.operators).isEquals();
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return new HashCodeBuilder(23, 87)
+                .appendSuper(super.hashCode())
+                .append(defaultOperator).append(operators)
+                .toHashCode();
+        }
+
+        @Override
+        public String toString()
+        {
+            return new ToStringBuilder(this)
```

Comment:
Also, aren't we supposed to use XWikiToStringBuilder?

Reply:
Yep, see https://dev.xwiki.org/xwiki/bin/view/Community/CodeStyle/JavaCodeStyle/#HEquals2FHashCodeandToStringimplementations